include/aws/s3/private/s3_auto_ranged_put.h

```diff
@@ -45,6 +45,8 @@ struct aws_s3_auto_ranged_put {
          * Throughout codebase 0 based part numbers are usually referred to as part index.
          */
         uint32_t next_part_number;
+
+        struct aws_array_list buffer_pool;
```

Comment:
Seems like we should use linked_list instead as we keep push/pop.

Reply:
array_list is fine if this thing is only a few dozen in size. simpler to use because we don't need to allocate a unique type with a node in it